### PR TITLE
Protect against files with zero pre-selected events

### PR DESCRIPTION
**Pull request by IzaakWN**
This PR adds a simple if-statement to protect to the case were zero events were preselected.

I ran into this error for several data files for which no events were in the good event list of the golden JSON, leading to failing jobs (`self.noOut` set to `True`). Otherwise I have to manually filter the input file list for the jobs.

### CODE REVIEW

**Comment by clelange** [comment]
Thanks! So https://github.com/cms-nanoAOD/nanoAOD-tools/pull/305/ didn't solve this issue but this is about empty trees?

**Comment by IzaakWN** [comment]
Thanks for the follow up.

No, I have the latest version and my jobs crash for files with zero pre-selected events because `outTree` is `None`.

I also use `noOut` set to true, otherwise, if `noOut` is false it would still fail few lines down when trying to write:
https://github.com/cms-nanoAOD/nanoAOD-tools/pull/306/files#diff-7a03b4209a618c9eb78be60742e95aa26c10bcdb784ef7f22b8b37550a79ff9eL247

Not sure if we prefer a totally empty output file?

File: python/postprocessing/framework/postprocessor.py [hunk 1]
```diff
@@ -238,7 +238,7 @@ def run(self):
                     eventRange=eventRange, maxEvents=self.maxEntries
                 )
                 print('Processed %d preselected entries from %s (%s entries). Finally selected %d entries' % (nall, fname, nEntries, npass))
-            else:
+            elif outTree!=None:
```
Comment (meisonlikesicecream) [inline python/postprocessing/framework/postprocessor.py:241]: `elif outTree is not None:`? :-)

**Comment by meisonlikesicecream** [comment]
> @meisonlikesicecream let us know if you have any comments and/or you're OK with this, too, mainly to avoid duplications. Then I would merge.

@clelange I'm OK, I always have an output tree, so this shouldn't affect me

File: python/postprocessing/framework/postprocessor.py (hunk 1 above)
Comment (IzaakWN) [reply python/postprocessing/framework/postprocessor.py]: Done. The rest rest of the code indeed uses `is`.

**Comment by IzaakWN** [comment]
I implemented @meisonlikesicecream's style suggestion, and sneaked in small cosmetic change for `haddnano.py` (adding spaces in print out).

**Review by meisonlikesicecream — APPROVED** [review]

**Comment by meisonlikesicecream** [comment]
Aww I can't review it, but looks good to me (nice with the spaces 😄)

**Review by clelange — APPROVED** [review]

**Pull request merged**